plugins/justify/plugin.js

```diff
@@ -178,8 +188,18 @@
 
 		refresh: function( editor, path ) {
 			var firstBlock = path.block || path.blockLimit;
-
-			this.setState( firstBlock.getName() != 'body' && getAlignment( firstBlock, this.editor.config.useComputedState ) == this.value ? CKEDITOR.TRISTATE_ON : CKEDITOR.TRISTATE_OFF );
+			var name = firstBlock.getName();
+
+			// Check if we can give to element proper class, style or current path is body (#16808).
+			// IE8, after focus and refresh return path with 'body'
+			if ( ( this.cssClassName && editor.activeFilter.check( name+'('+this.cssClassName+')' ) ) ||
+				editor.activeFilter.check( name+'{text-align}' ) ||
+				name == 'body' )
```

Comment:
This should work as long as the classic, iframe based editor is used. However in div-based editor this check will be always `false`. Therefore I'd check if the `firstBlock` is not the same node as editable.